Application/Frontend/components/Server/Member_List.tsx

```diff
@@ -1,35 +1,46 @@
 import React, { useEffect, useState } from 'react';
-import { Avatar, Group, Text, Stack, Paper, Button, Menu, rem } from '@mantine/core';
+import { Avatar, Group, Text, Stack, Paper, Button, Menu, rem, TextInput, Modal } from '@mantine/core';
 import { IconSettings, IconMessageCircle, IconPhoto, IconSearch, IconArrowsLeftRight, IconTrash, IconPlus, IconUserUp } from '@tabler/icons-react';
 import { useUser } from '@clerk/nextjs';
+import { channel } from 'diagnostics_channel';
+import { createHash } from 'crypto';
+import { IntegerType } from 'mongodb';
+import { useDisclosure } from '@mantine/hooks';
+import { showNotification } from '@mantine/notifications';
 
-export function MemberList({isAdmin}: any) {
+const generateHash = (input: string) => {
```

Comment:
This has function should exist in a utilities file as it is used in two places (MessagingInterface and your component). That means we have duplicated code. This is something I will patch.